content/2017.04.getting-started-with-elm.rst

```diff
@@ -0,0 +1,487 @@
+
+
+How is Elm different?
+=====================
+
+First of all, using Elm, the two lines it takes to load your app is
+all the JavaScript you will write. Your application will be written in
+Elm and compiled into JavaScript.
+
+All the mistakes you will do will be caught at compilation, which
+means that you will not have errors when running your app in the
+browser.
+
+Because you will have a compilation step, Elm will force you to handle
+every possible case of your model state. It will make sure you covered
+them on the rendering side.
+
+The tooling is great. ``elm-live`` will automatically update the
+browser code to reflect your code update in realtime. You have a
+debugger that will show you the list of events and show you the state
+of your application at a given time. You can even go back in time and
+replay events. ``elm-format`` will also format your code automatically
+to avoid fighting over coding styles.
+
+
+Getting started
+===============
+
+Installing Elm
+--------------
+
+First of all install ``elm`` and ``elm-format`` using npm the node
+package manager:
+
+.. code-block:: bash
+
+    npm install -g elm elm-format
+
+What is npm? How to install it?
+-------------------------------
+
+npm is the node package manager. To install it, you might want to use
+`nvm <https://github.com/creationix/nvm#install-script>`_.
+
+Once nvm is installed, you can use::
+
+    nvm install node
+
+And then you wil be able to run the previous ``npm`` command.
+
+Looking for an IDE?
+-------------------
+
+Have a look at `Atom <https://atom.io/>`. By default it will run
+``elm-format`` each time you save your file which is really handy.
+
+Take a few moments to make sure your text editor is configured to work
+friendly with Elm files.
+
+
+Starting your first project
+===========================
+
+No need for a boilerplate here, you can just start by running
+``elm-package install`` and approve the plan:
+
+.. code-block:: bash
+
+    ~/tutorial$ elm-package install
+
+    Some new packages are needed. Here is the upgrade plan.
+
+      Install:
+        elm-lang/core 5.1.1
+        elm-lang/html 2.0.0
+        elm-lang/virtual-dom 2.0.4
+
+    Do you approve of this plan? [Y/n] Y
+    Starting downloads...
+
+      ● elm-lang/html 2.0.0
+      ● elm-lang/virtual-dom 2.0.4
+      ● elm-lang/core 5.1.1
+
+    Packages configured successfully!
+
+You now have a ``elm-package.json`` in your project as well as a
+``elm-stuff/`` directory that contains libraries ``elm-package``
+installed.
+
+.. code-block:: bash
+
+    ~/tutorial$ tree -L2
+	.
+    ├── elm-package.json
+    └── elm-stuff
+        ├── exact-dependencies.json
+        └── packages
+
+    2 directories, 2 files
+
+
+Creating your first file
+========================
+
+No need for a boilerplate here, to get started you can simply create a
+new file named ``Main.elm`` with the following:
+
+.. code-block:: elm
+
+    import Html
+    main = Html.text "Hello world"
```

Comment:
Maybe state that Elm benefits from a full featured module system, with a broad ecosystem of external packages available, and that `Html` is part of the core.